clients/src/main/java/org/apache/kafka/common/header/Headers.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.header;
+
+public interface Headers extends Iterable<Header> {
+    
+    /**
+     * Adds a header (key inside), to the end, returning if the operation succeeded.
+     * 
+     * @param header the Header to be added
+     * @return this instance of the Headers, once the header is added.
+     * @throws IllegalStateException is thrown if headers are in a read-only state.
+     */
+    Headers add(Header header) throws IllegalStateException;
```

Comment:
If we added another method `add(String key, byte[] value)`, would there be any need to expose a concrete implementation of `Header`?